src/CalcViewModel/Common/LocalizationStringUtil.h

```diff
@@ -9,50 +9,87 @@ namespace CalculatorApp
 {
     namespace Common
     {
-        class LocalizationStringUtil
+         class LocalizationStringUtilInternal
         {
         public:
-            static std::wstring GetLocalizedString(const wchar_t* pMessage, ...)
-            {
+            static Platform::String ^ GetLocalizedString(Platform::String ^ pMessage, ...) {
                 std::wstring returnString = L"";
                 const UINT32 length = 1024;
                 std::unique_ptr<wchar_t[]> spBuffer = std::unique_ptr<wchar_t[]>(new wchar_t[length]);
                 va_list args = NULL;
                 va_start(args, pMessage);
-                DWORD fmtReturnVal = FormatMessage(FORMAT_MESSAGE_FROM_STRING, pMessage, 0, 0, spBuffer.get(), length, &args);
+                DWORD fmtReturnVal = FormatMessage(FORMAT_MESSAGE_FROM_STRING, pMessage->Data(), 0, 0, spBuffer.get(), length, &args);
                 va_end(args);
 
                 if (fmtReturnVal != 0)
                 {
-                    returnString = spBuffer.get();
+                    return ref new Platform::String(spBuffer.get());
+                }
+                else
+                {
+                    return ref new Platform::String();
                 }
-
-                return returnString;
             }
+        };
 
-            template <typename... T>
-            static Platform::String^ GetLocalizedNarratorAnnouncement(Platform::String^ resourceKey, Platform::String^& formatVariable, T*... params)
-            {
-                EnsureInitialization(resourceKey, formatVariable);
-                return StringReference(GetLocalizedString(formatVariable->Data(), params...).c_str());
-            }
+    public
+        ref class LocalizationStringUtil sealed
+        {
+        public:
+            static Platform::String
+                ^ GetLocalizedString(Platform::String ^ pMessage)
+                {
+                    return LocalizationStringUtilInternal::GetLocalizedString(pMessage);
+                }
 
-        private:
-            static void EnsureInitialization(Platform::String^ resourceKey, Platform::String^& formatVariable)
-            {
-                if (resourceKey == nullptr || resourceKey->IsEmpty())
+                static Platform::String
```

Comment:
nit: all methods in this class have weird formatting

Reply:
A good example of how broken is clang with C++/CX  (especially with `^`) 😫